src/main/java/filter/Parser.java

```diff
@@ -181,6 +181,8 @@ private FilterExpression parseSeparatedContent(Supplier<FilterExpression> conten
     private FilterExpression parseQualifierContent(QualifierType type) {
         if (type == QualifierType.SORT) {
             return parseSortKeys();
+        } else if (isCompoundId(type, lookAhead())) {
```

Comment:
Okay then. Could we check the type outside, like with `sort`? So it's more explicit which qualifiers we do this for.